### PR TITLE
Feat : Labels

### PR DESCRIPTION
**Pull request by shlok30**
Hi Team,

I have added 'Labels' feature to my notes app.

Users can create new labels and select which of them should apply to the current note

Demo Link :  https://water-notes-3ud6e9fhp-shlok30.vercel.app/login

Video Link : 

https://user-images.githubusercontent.com/60795588/174425647-16cc33ea-014e-4a4a-9cad-8ae4de6c3ee8.mov

Can you please review my code?

Thanks

### CODE REVIEW

File: src/components/modal-components/modal-elements.js [hunk 1]
```diff
@@ -0,0 +1,31 @@
+import { useNotes } from "../../context/notes-context"
+
+const ModalElements = ({label}) => {
+
+    const {notesState : {labels},notesDispatch} = useNotes()
+
+    console.log("Labels from notes state ",labels)
```
Comment (nikhil-sharma-nks) [inline src/components/modal-components/modal-elements.js:7]: remove console log from here

File: src/components/modal-components/modal-elements.js [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+        else{
+            tempCopyOfLabels = labels.filter(labelElement => labelElement !== label)
+            notesDispatch({type : "LABELS", payload : tempCopyOfLabels})
+
+        }
+    }
+
+    return(
+        <div className="flex gap-m align-center">
+            <input type = "checkbox" onChange={labelHandler} checked = {labels.includes(label)} />
```
Comment (nikhil-sharma-nks) [inline src/components/modal-components/modal-elements.js:25]: good implementation of includes method here

File: src/general-functions/renderModalElements.js [hunk 3]
```diff
@@ -0,0 +1,5 @@
+import ModalElements from "../components/modal-components/modal-elements";
+
+const renderModalElements = (labels) => labels.map(label => <ModalElements label = {label} />)
```
Comment (nikhil-sharma-nks) [inline src/general-functions/renderModalElements.js:3]: it's a good practice to add key when rendering multiple components with map method

File: src/pages/homepage.js [hunk 4]
```diff
@@ -1,17 +1,19 @@
...
 import NoteEditor from '../components/note-editor'
 import renderNoteCards from "../general-functions/renderNoteCards"
+import LabelsModal from "../components/labels-modal"
 
 const Homepage = () => {
 
-    const {userState : {isLoggedIn,notes}} = useUser()
+    const {userState : {isLoggedIn,notes,isModalActive}} = useUser()
 
     console.log(isLoggedIn)
```
Comment (nikhil-sharma-nks) [inline src/pages/homepage.js:10]: remove console log from here

**Review by nikhil-sharma-nks — APPROVED** [review]
Looks good to merge. You've good code structure.

File: src/reducers/notesReducer.js [hunk 5]
```diff
@@ -6,11 +6,13 @@ const notesReducer = (state,{type,payload}) => {
...
-            return {...state, title : "" , body : "", colour : "", priority : "0"}
+            return {...state, title : "" , body : "", colour : "", priority : "0", labels : []}
         case "COLOUR": 
             return {...state,colour : payload}
         case "PRIORITY":
             return {...state,priority : payload}
+        case "LABELS":
+            return {...state, labels : [...payload]}
         default:
             return
```
Comment (ranitasahaa1230) [inline src/reducers/notesReducer.js:17]: Instead of returning state in the default, you can throw an error over here

**Review by ranitasahaa1230 — APPROVED** [review]
Remove all the console.logs
Clean UI
Rest, looks good

**Pull request merged**